Submitted By:            Matt Burgess <matthew_at_linuxfromscratch_dot_org>
Date:                    2010-05-19
Initial Package Version: 0.18
Upstream Status:         Submitted
Origin:                  Matt Burgess
Description:             Skip a glade-related test that incorrectly gets run
                         when glade support hasn't been compiled in and
                         subsequently fails.

diff -Naur gettext-0.18.orig/gettext-tools/tests/xgettext-glade-4 gettext-0.18/gettext-tools/tests/xgettext-glade-4
--- gettext-0.18.orig/gettext-tools/tests/xgettext-glade-4	2009-06-28 19:44:04.000000000 +0000
+++ gettext-0.18/gettext-tools/tests/xgettext-glade-4	2010-05-19 18:31:01.433048303 +0000
@@ -5,6 +5,20 @@
 tmpfiles=""
 trap 'rm -fr $tmpfiles' 1 2 3 15
 
+tmpfiles="$tmpfiles empty.glade"
+cat <<EOF > empty.glade
+<?xml version="1.0"?>
+<GTK-Interface/>
+EOF
+
+tmpfiles="$tmpfiles xg-gl-4.pot"
+: ${XGETTEXT=xgettext}
+${XGETTEXT} -o xg-gl-4.pot empty.glade 2>/dev/null
+test $? = 0 || {
+  echo "Skipping test: xgettext was built without Glade support"
+  rm -fr $tmpfiles; exit 77
+}
+
 tmpfiles="$tmpfiles xg-gl-4.glade"
 cat <<EOF > xg-gl-4.glade
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
